app/controllers/publish/courses/funding_type_controller.rb

```diff
@@ -45,7 +38,34 @@ def update
       def funding_type_params
         return {} if params[:publish_course_funding_form].blank?
 
-        params.require(:publish_course_funding_form).permit(:funding_type)
+        params.require(:publish_course_funding_form).permit(:funding_type, :previous_tda_course)
+      end
+
+      def handle_valid_form
+        if previous_tda_course?
+          process_previous_tda_course
+        else
+          redirect_to next_path
+        end
+      end
+
+      def handle_invalid_form
+        @errors = @course_funding_form.errors.messages
+        render :edit
+      end
+
+      def previous_tda_course?
+        params[:publish_course_funding_form][:previous_tda_course] == 'true'
```

Comment:
Just double checking if the feature is off 

`params[:publish_course_funding_form]` will be `{ ... }` and `params[:publish_course_funding_form][:previous_tda_course]` will be false.

Reply:
With the feature flag on: 

```
params[:publish_course_funding_form]
{"funding_type"=>"fee", "previous_tda_course"=>"true"}

params[:publish_course_funding_form][:previous_tda_course]
"true"
```

With the feature flag off you will not even be able to get to this update action because you won't have the option to select a TDA course in the first place.